tests/components/yardian/test_sensor.py

```diff
@@ -0,0 +1,156 @@
+"""Tests for Yardian sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import UTC, datetime
+from unittest.mock import patch
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.yardian.const import DOMAIN
+from homeassistant.components.yardian.coordinator import (
+    YardianCombinedState,
+    YardianUpdateCoordinator,
+)
+from homeassistant.components.yardian.sensor import (
+    _rain_delay_value,
+    _water_hammer_duration_value,
+    _zone_delay_value,
+)
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensor_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    mock_config_entry: MockConfigEntry,
+    mock_yardian_client: object,
+) -> None:
+    """Snapshot all Yardian sensor entities."""
+
+    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_diagnostic_sensors(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_yardian_client: object,
+) -> None:
+    """Diagnostic sensors are disabled by default but report values when enabled."""
+
+    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+    ent_reg = er.async_get(hass)
```

Comment:
```suggestion
    mock_yardian_client: object,
    entity_registry: er.EntityRegistry,
) -> None:
    """Diagnostic sensors are disabled by default but report values when enabled."""

    with patch("homeassistant.components.yardian.PLATFORMS", [Platform.SENSOR]):
        await setup_integration(hass, mock_config_entry)
```
It can just be loaded as test fixture